src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+                    reagent.normalize(archive, logger)
+                reagents.append(reagent)
+            self.reaction_conditions.reagents = reagents
+
+            if self.reactor_filling is not None:
+                if (
+                    self.reaction_conditions.set_total_flow_rate is not None
+                    and self.reactor_filling.catalyst_mass is not None
+                    and self.reaction_conditions.weight_hourly_space_velocity is None
+                ):
+                    self.reaction_conditions.weight_hourly_space_velocity = (
+                        self.reaction_conditions.set_total_flow_rate
+                        / self.reactor_filling.catalyst_mass
+                    )
+
+    def return_conversion_results(
+        self, archive: 'EntryArchive', logger: 'BoundLogger'
+    ) -> None:
+        """
+        This function returns the conversion results of the reactants for the results
+        section of the archive.
+        It checks if the the name of the reactant is not in the list of the inert gases
+        and if the name is in the list of the reaction_conditions.reagents, it will try
+        to replace the name of the reactant with the IUPAC name of the reagent.
+
+        return: a list of the reactants with the conversion results.
+        """
+        if self.results[0].reactants_conversions is not None:
+            conversions_results = []
+            for i in self.results[0].reactants_conversions:
+                if i.name in ['He', 'helium', 'Ar', 'argon', 'inert']:
+                    continue
+                else:
+                    for j in self.reaction_conditions.reagents:
+                        if i.name == j.name:
+                            if j.pure_component.iupac_name is not None:
+                                i.name = j.pure_component.iupac_name
+                            if i.gas_concentration_in is None:
+                                i.gas_concentration_in = j.gas_concentration_in
+                            react = Reactant(
+                                name=i.name,
+                                conversion=i.conversion,
+                                gas_concentration_in=i.gas_concentration_in,
+                                gas_concentration_out=i.gas_concentration_out,
+                            )
+                            conversions_results.append(react)
+                            if (
+                                np.allclose(
+                                    i.gas_concentration_in, j.gas_concentration_in
+                                )
+                                is False
+                            ):
+                                logger.warn(f"""Gas concentration of '{i.name}' is not
+                                            the same in reaction_conditions and
+                                            results.reactants_conversions.""")
+        return conversions_results
```

Comment:
Firstly, I think this method will fail if `reactants_conversions` is `None` as you are then trying to return `conversions_results` before they are defined. Secondly, the method probably works but looks very indented so I will try to give some tips on how it could be simplified :)
1. Negate the first check as I mentioned before
2. There is no need for `else:` after a `continue` or `return` in the the `if:` before
3. Again negate the check and continue
4. In Python the convention is to use the `not` keyword instead of `is False`.
This is not tested but *should* theoretically work ;)
```suggestion
        if self.results[0].reactants_conversions is None:
            return []
        conversions_results = []
        for i in self.results[0].reactants_conversions:
            if i.name in ['He', 'helium', 'Ar', 'argon', 'inert']:
                continue
            for j in self.reaction_conditions.reagents:
                if i.name != j.name:
                    continue
                if j.pure_component.iupac_name is not None:
                    i.name = j.pure_component.iupac_name
                if i.gas_concentration_in is None:
                    i.gas_concentration_in = j.gas_concentration_in
                react = Reactant(
                    name=i.name,
                    conversion=i.conversion,
                    gas_concentration_in=i.gas_concentration_in,
                    gas_concentration_out=i.gas_concentration_out,
                )
                conversions_results.append(react)
                if not np.allclose(i.gas_concentration_in, j.gas_concentration_in):
                    logger.warn(f"""Gas concentration of '{i.name}' is not
                                the same in reaction_conditions and
                                results.reactants_conversions.""")
        return conversions_results
```